### PR TITLE
Add some notes on selection behaviours for review

### PR DESCRIPTION
**Pull request by bz2**
Not the most coherent yet as there's several things we don't really have good names for, but tries to summarise a few of the challenges with picking what to show when a selection is made.
[Jira ticket](https://visual-meaning.atlassian.net/browse/SMP-2728)

### CODE REVIEW

File: comments/framing_for_selection.md [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+The presented selection should be clear and meaningful. That means it should ideally have visible on screen all of the content related to the selection, and some of the surrounding context from the environment, such as the parent, and siblings. Content should not be obscured by other elements of the platform. For container-type items, what is contained within them should be clear, even if all descendants are not shown at that detail level. There's an intersection between map design, ontology, and behaviours derived from the model.
+
+
+## Existing behaviours
+
+Currently there are three distinct cases to consider:
+* New map load with selection set
+* Select new item from link outside the map
+* Select new item on the map
```
Comment (rosi-grafixoft) [inline comments/framing_for_selection.md:27]: is this the selection by just clicking on the map or you mean something else

File: comments/framing_for_selection.md [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+## Expectations
+
+The presented selection should be clear and meaningful. That means it should ideally have visible on screen all of the content related to the selection, and some of the surrounding context from the environment, such as the parent, and siblings. Content should not be obscured by other elements of the platform. For container-type items, what is contained within them should be clear, even if all descendants are not shown at that detail level. There's an intersection between map design, ontology, and behaviours derived from the model.
+
+
+## Existing behaviours
+
+Currently there are three distinct cases to consider:
+* New map load with selection set
```
Comment (rosi-grafixoft) [inline comments/framing_for_selection.md:25]: is this when for example you switch to other map via the SIP

File: comments/framing_for_selection.md [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+* Select new item from link outside the map
```
Comment (rosi-grafixoft) [inline comments/framing_for_selection.md:26]: sorry but map and map can mean different. You mean if I just load a URL from email or I'm on Org map but need to go to Activity map

File: comments/framing_for_selection.md [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+The new map load does not animate, but does show the constrained portion of the map and any selection effects.
+
+Selecting by a navigation link causes an animation from the current map location, ending up at the same position as a new load. It's similar to the leaflet flyto effect in that it includes some zoom as well as pan, but does not have an adaptive speed based on distance at present.
+
+Selecting by something on the map, a marker or label, does not cause any animation or change to the position of the map. (Note that web mapping platforms do tend to animate in this case, at least some of the time, for example if selecting a placename, but not if selecting a marker for a location.)
+
+
+## Bounds
+
+Bounds for an item to zoom to is currently often a rough heuristic based on the vector export, which is a rectangle drawn around all of the content at that semantic level. This is a reasonable 80% solution, without a lot of manual work to add outlines in the illustrator file, but can have edge case surprising behaviours.
```
Comment (alightwing) [inline comments/framing_for_selection.md:38]: Are the bounds/vector export parameters and how they end up displaying on the platform documented anywhere? I spent some time with Mark last week trying to figure out how a consultant's intent in Illustrator was expressed on the platform, and didn't get very far.

Comment (tracivar) [inline comments/framing_for_selection.md:38]: One problem of bounds is that it is connected to the name of the item in the model, so if there are multiple items sharing the same name, their corresponding bounds can be wrong

File: comments/framing_for_selection.md [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+
+## What should change
+
+Bounds are not intrinsic to an item, they are specific to an item in a lens. To update this, we need an equivalently good heuristic for what to show on screen for any selection.
```
Comment (tracivar) [inline comments/framing_for_selection.md:43]: We want to zoom to the coords of the label, and on the first detail level they appear on? Currently this behaviour is inconsistent for desktop and mobile view, sometimes too zoomed in or sometimes too zoomed out. We should make sure each item's lens' details level is sensible?

**Review by kisoso — APPROVED** [review]
Thanks.

This doc should be a good base for future development.
Some points may be reclarified to be for desired state or current state that may change with time (as noted in the comments).

**Comment by bz2** [comment]
Will want adopting if we come back around on selection ideas, and turning into something more actionable.

**Pull request merged**